app/lib/three_scale/policies/specification.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  module Policies
+    class Specification
+      extend ActiveModel::Naming
+      extend ActiveModel::Translation
+
+      def initialize(body)
+        @errors = ActiveModel::Errors.new(self)
+
+        begin
+          @doc = JSON.parse(body.to_s)
+        rescue JSON::ParserError
+          @errors.add(:base, :invalid_json)
+        end
+
+        @doc ||= {}
+      end
+
+      attr_reader :errors, :doc
+
+      def valid?
+        return false if errors.any?
```

Comment:
If it were actually an ActiveModel we would expect `valid?` to run all the validations repeatedly whenever we call it. Maybe `JSON::ParserError` should be internally stored separately.

Reply:
Now I see why. If there's a `JSON::ParserError` in the initialization, then there's no way to recover from it because `body` is private.

Reply:
Probably assuming too much from the passed string.
Why not separate the responsibility in 2 and only accept a Hash as `#initialize` argument?